src/App.tsx

```diff
@@ -1,27 +1,95 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
-import { GoodsList } from './GoodsList';
+import classNames from 'classnames';
+import { GoodsList } from './components/GoodsList';
+import { get5First, getAll, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+import { LoadingError } from './components/LoadingError';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [loading, setLoad] = useState({
+    all: false,
+    five: false,
+    red: false,
+  });
```

Comment:
Interesting idea, but as for me it's strange that we run query before we get previous query result
I guess will be more logical to disable all buttons while something is loading